embedded/sql/stmt.go

```diff
@@ -1648,6 +1718,116 @@ func (v *SysFn) selectorRanges(table *Table, asTable string, params map[string]i
 	return nil
 }
 
+type Cast struct {
+	val ValueExp
+	t   SQLValueType
+}
+
+func (c *Cast) getConverter(src, dst SQLValueType) func(TypedValue) (TypedValue, error) {
+	if dst == TimestampType {
+
+		if src == IntegerType {
```

Comment:
wondering is casting from typed null values e.g. a stmt involving a varchar column and some rows having null values...

Reply:
Added handling of null values (there was a panic if cast was done with a null str or null integer)

Reply:
it might be also better to return an error when there is no converter for the specified types?

Reply:
Updated the `getConverter` method - now errors are much more descriptive.

Reply:
Regarding conversion from NULL to NULL timestamp - I wanted to skip this in this version because that comes with a lot of weird corner cases because NULL can be of arbitrary type, let's start with a simple solution that works in most practical use-cases and extend it later.